controllers/cluster_controller_postapply.go

```diff
@@ -0,0 +1,199 @@
+package controllers
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"io"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilyaml "k8s.io/apimachinery/pkg/util/yaml"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/controllers/remote"
+	"sigs.k8s.io/cluster-api/util/secret"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	errUninitializedControlPlane = errors.New("control plane is uninitialized")
+)
+
+func (r *ClusterReconciler) reconcilePostApply(ctx context.Context, cluster *clusterv1.Cluster) error {
+	logger := r.Log.WithValues("cluster", cluster.Name, "namespace", cluster.Namespace)
+
+	if !cluster.Status.ControlPlaneInitialized {
+		return errUninitializedControlPlane
+	}
+
+	c, err := remote.NewClusterClient(ctx, r.Client, cluster, r.scheme)
+	// Failed to get remote cluster client: Kubeconfig secret may be missing for the cluster.
+	if err != nil {
+		return err
+	}
+
+	addons := cluster.Spec.PostApplyAddons
+	for _, addon := range addons {
```

Comment:
Would there be value in aggregating the errors we encounter in this loop and returning that back instead of swallowing them all? Something similar to what we are doing here: https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/cluster_controller.go#L150-L174